src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -19,13 +19,36 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    /** Indicates if the command is a group-related command. */
+    private final boolean isGroupCommand;
+
     /**
-     * Constructs a {@code CommandResult} with the specified fields.
+     * Constructs a {@code CommandResult} with the specified feedback, help, and exit fields.
+     *
+     * @param feedbackToUser The feedback message to be shown to the user.
+     * @param showHelp       Whether the help information should be shown to the user.
+     * @param exit           Whether the application should exit after this command.
      */
     public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
         this.feedbackToUser = requireNonNull(feedbackToUser);
         this.showHelp = showHelp;
         this.exit = exit;
+        this.isGroupCommand = false;
+    }
+
+    /**
+     * Constructs a {@code CommandResult} with the specified feedback, help, exit, and group command fields.
+     *
+     * @param feedbackToUser The feedback message to be shown to the user.
+     * @param showHelp       Whether the help information should be shown to the user.
+     * @param exit           Whether the application should exit after this command.
+     * @param isGroupCommand Whether this command pertains to group-related operations.
+     */
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean isGroupCommand) {
+        this.feedbackToUser = requireNonNull(feedbackToUser);
+        this.showHelp = showHelp;
+        this.exit = exit;
+        this.isGroupCommand = isGroupCommand;
```

Comment:
Would doing
``` Java
public CommandResult(String feedbackToUser, boolean isGroupCommand)` {
  ...
}
```
be better? Otherwise, I think it will be better to refactor the above method.